Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Overlay/SendOverlay.hs

```diff
@@ -38,72 +42,95 @@ sendOverlayToDriver ::
 
-getFirstNPaymentPendingDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Integer -> m [Id DP.Person]
-getFirstNPaymentPendingDriverIds merchantId num = Hedis.lRange (makePaymentPendingDriverIdsKey merchantId) 0 (num -1)
+getFirstNSendOverlaySchedulerDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> Integer -> m [Id DP.Person]
+getFirstNSendOverlaySchedulerDriverIds merchantId jobId num = Hedis.lRange (makeSendOverlaySchedulerDriverIdsKey merchantId jobId) 0 (num -1)
 
-deleteNPaymentPendingDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Integer -> m ()
-deleteNPaymentPendingDriverIds merchantId num = Hedis.lTrim (makePaymentPendingDriverIdsKey merchantId) num (-1)
+deleteNSendOverlaySchedulerDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> Integer -> m ()
+deleteNSendOverlaySchedulerDriverIds merchantId jobId num = Hedis.lTrim (makeSendOverlaySchedulerDriverIdsKey merchantId jobId) num (-1)
 
-addPaymentPendingDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> NonEmpty (Id DP.Person) -> m ()
-addPaymentPendingDriverIds merchantId = Hedis.rPush (makePaymentPendingDriverIdsKey merchantId)
+addSendOverlaySchedulerDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> NonEmpty (Id DP.Person) -> m ()
+addSendOverlaySchedulerDriverIds merchantId jobId = Hedis.rPush (makeSendOverlaySchedulerDriverIdsKey merchantId jobId)
 
-makePaymentPendingDriverIdsKey :: Id DM.Merchant -> Text
-makePaymentPendingDriverIdsKey merchantId = "SendOverlayScheduler:PaymentPendingDriverIds:merchantId-" <> merchantId.getId
+makeSendOverlaySchedulerDriverIdsKey :: Id DM.Merchant -> Id AnyJob -> Text
+makeSendOverlaySchedulerDriverIdsKey merchantId jobId = "SendOverlayScheduler:merchantId-" <> merchantId.getId <> ":jobId-" <> jobId.getId
 
-getBatchedDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> m [Id DP.Person]
-getBatchedDriverIds merchantId = do
-  driverIdsLength <- getPaymentPendingDriverIdsLength merchantId
+getBatchedDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> DOverlay.OverlayCondition -> m [Id DP.Person]
+getBatchedDriverIds merchantId jobId condition = do
+  driverIdsLength <- getSendOverlaySchedulerDriverIdsLength merchantId jobId
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
   when (driverIdsLength < 1) do
-    drivers <- QDI.fetchAllDriversWithPaymentPending merchantId
-    whenJust (nonEmpty (drivers <&> (.driverId))) $ addPaymentPendingDriverIds merchantId
-  batchedDriverIds <- getFirstNPaymentPendingDriverIds merchantId 50
-  deleteNPaymentPendingDriverIds merchantId 50
+    driverIds <- case condition of
+      DOverlay.PaymentOverdueGreaterThan _ -> QDI.fetchAllDriversWithPaymentPending merchantId <&> (<&> (.driverId))
```

Comment:
This will send all the driverIds with any pending payment

Reply:
yes, ideally it will fetch all the driverIds with paymentPending flag set to true. ( in driver information table ) - we are caching these driverIds in rediscache

Reply:
But where is the check of greater than limit or in b/w l and r?